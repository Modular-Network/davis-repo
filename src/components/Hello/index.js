import React from 'react';
import Image from 'react-image-resizer';
import blossom from 'assets/blossom_logo_tm.svg';
import styles from './styles.css';

/*TODO: Add transtion to fade up and in on load and fade down and out on exit*/

const Hello = ({movePage}) => {
  return (
    <div className={styles.base}>
      <div className={styles.hello}>Hello,</div>
      <div className={styles.blossom}>Welcome to </div>
      <div className={styles.logo}>{<Image src={blossom} height={ 45 } width={ 200 }/>}</div>
      <div className={styles.lets} onClick={e => movePage(1) }>{"Let's get started â†’ "}</div>
    </div>
  )
}

export default Hello;
