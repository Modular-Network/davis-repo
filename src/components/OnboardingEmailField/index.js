import React from 'react';
import styles from './styles.css';
import { TextValidator, ValidatorForm } from 'react-material-ui-form-validator';

const OnboardingEmailField = ({ onChange, text, type, value, changeSection }) => (
    <div className={styles.base}>
        <ValidatorForm onSubmit={() => {}} debounceTime={1000}>
            <TextValidator
                onChange={(e) => onChange({email: e.target.value, error: false})}
                hintText={value}
                floatingLabelText={text}
                type={type}
                fullWidth={true}
                underlineStyle={{ borderColor: 'white' }}
                underlineFocusStyle={{ borderColor: '#FFB087', opacity: '0.8' }}
                floatingLabelStyle={{ color: 'white', fontSize: '3.6rem', top: '-2rem' }}
                floatingLabelFocusStyle={{ color: 'white', fontSize: '2.4rem' }}
                inputStyle={{ color: 'white', fontSize: '6rem', top: '-2rem', fontWeight: '300', width: '90%' }}
                errorStyle={{ color: 'yellow', position: 'absolute', bottom: '-15px'}}
                value={value}
                name="email"
                validators={['required', 'isEmail']}
                errorMessages={['this field is required', 'email is not valid']}
                onKeyPress={e => {
                    if ((e.key === 'Enter') && (value !== '')) {
                        changeSection()
                    }
                }}
            />
        </ValidatorForm>
        <div className={value === '' ? styles.dark : styles.circle} onClick={value === '' ? null : changeSection}></div>
    </div>
);

export default OnboardingEmailField;
