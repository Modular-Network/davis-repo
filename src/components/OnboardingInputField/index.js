import React from 'react';
import TextField from 'material-ui/TextField';
import styles from './styles.css';

/*TODO: Add icon in circles and opacity transitions*/

const OnboardingInputField = ({ onChange, text, type, value, action }) => (
  <div className={styles.base}>
    <TextField
      onChange={(e) => onChange({username: e.target.value, error: false})}
      hintText={value}
      floatingLabelText={text}
      type={type}
      fullWidth={true}
      underlineStyle={{ borderColor: 'white' }}
      underlineFocusStyle={{ borderColor: '#FFB087', opacity: '0.8'}}
      floatingLabelStyle={{ color: 'white', fontSize: '3.6rem', top: '-2rem'}}
      floatingLabelFocusStyle={{ color: 'white', fontSize: '2.4rem'}}
      inputStyle={{ color: 'white', fontSize: '6rem', top: '-2rem', fontWeight: '300', width: '90%'}}
      value={value}
      onKeyPress={e => {
        if ((e.key === 'Enter') && (value !== '')) {
          action()
        }
      }}
    />
    <div className={value === '' ? styles.dark : styles.circle} onClick={value === '' ? null : action}></div>
  </div>
);

export default OnboardingInputField;
