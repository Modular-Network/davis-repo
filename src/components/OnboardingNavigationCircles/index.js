import React from 'react';
import styles from './styles.css';

const OnboardingNavigationCircles = ({pages, selected}) => {
  let circles = [];
  for(let i = 0; i <= pages; i++) {
    circles.push(<div key={i} className={selected === i ? styles.lit : styles.grey} />);
  }
  return (
    <div className={styles.base}>
      {circles}
    </div>
  )
}

export default OnboardingNavigationCircles;
