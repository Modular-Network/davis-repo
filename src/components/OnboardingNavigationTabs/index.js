import React from 'react';
import styles from './styles.css';

const getLine = (section) => {
  switch (section) {
    case 'user':
      return styles.lUser;
    case 'email':
      return styles.lEmail;
    case 'pass':
      return styles.lPass;
    case 'phone':
      return styles.lPhone;
    default:
      return 'error'
  }
}

const OnboardingNavigationTabs = ({ section, changeSection }) => {
  let lClass = getLine(section)

  return (
    <div>
      <div className={styles.base}>
        <div className={styles.user} onClick={e => changeSection('user')}>USERNAME</div>
        <div className={styles.email} onClick={e => changeSection('email')}>EMAIL ADDRESS</div>
        <div className={styles.pass} onClick={e => changeSection('pass')}>PASSWORD</div>
        <div className={styles.phone} onClick={e => changeSection('phone')}>PHONE NUMBER</div>
      </div>
      <div className={lClass}></div>
    </div>
  )
}

export default OnboardingNavigationTabs;
