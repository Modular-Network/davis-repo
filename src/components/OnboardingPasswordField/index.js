import React, { Component } from 'react';
import { TextValidator, ValidatorForm } from 'react-material-ui-form-validator';
import styles from './styles.css';

class OnboardingPasswordField extends Component {

  componentDidMount() {
    // custom rule will have name 'isPasswordMatch'
    ValidatorForm.addValidationRule('isPasswordMatch', (value) => {
      if (value !== this.props.passOne) {
        return false;
      }
      return true;
    });
  }

  render() {
    const { onChange, text, passOne, passTwo, changeSection } = this.props;

    return (
      <div className={styles.base}>
        <ValidatorForm onSubmit={() => {}}>
          <TextValidator
            onChange={(e) => onChange({password: e.target.value, error: false})}
            fullWidth={true}
            floatingLabelText={text}
            underlineStyle={{ borderColor: 'white' }}
            underlineFocusStyle={{ borderColor: '#FFB087', opacity: '0.8' }}
            floatingLabelStyle={{ color: 'white', fontSize: '3.6rem', top: '-2rem' }}
            floatingLabelFocusStyle={{ color: 'white', fontSize: '2.4rem' }}
            inputStyle={{ color: 'white', fontSize: '4rem', top: '-1rem' }}
            errorStyle={{ color: 'yellow', position: 'absolute', bottom: '-1rem' }}
            name="password"
            type="password"
            validators={['required', 'matchRegexp:(?=.{8,})(?=.*[A-Z])']}
            errorMessages={['this field is required', 'must contain a minimum of 8 characters and one uppercase character.']}
            value={passOne}
            className={styles.inputOne}
            onKeyPress={e => {
              if ((e.key === 'Enter') && (passOne !== '')) {
                changeSection()
              }
            }}
          />
          <div className={passOne === '' ? styles.dark : styles.circle}></div>
          <TextValidator
            onChange={(e) => onChange({secondPass: e.target.value, error: false})}
            name="repeatPassword"
            type="password"
            validators={['isPasswordMatch', 'required']}
            errorMessages={['password mismatch', 'this field is required']}
            fullWidth={true}
            underlineStyle={{ borderColor: 'white' }}
            underlineFocusStyle={{ borderColor: '#FFB087', opacity: '0.8' }}
            inputStyle={{ color: 'white', fontSize: '4rem', top: '-1rem' }}
            errorStyle={{ color: 'yellow', position: 'absolute', bottom: '-1rem'}}
            value={passTwo}
            className={styles.inputTwo}
            onKeyPress={e => {
              if ((e.key === 'Enter') && (passTwo !== '')) {
                changeSection()
              }
            }}
          />
        </ValidatorForm>
        <div className={passTwo === '' ? styles.dark : styles.circle} onClick={passTwo === '' ? null : changeSection}></div>
      </div>
    );
  }
}

export default OnboardingPasswordField