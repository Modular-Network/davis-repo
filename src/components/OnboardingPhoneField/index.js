import React from 'react';
import styles from './styles.css';
import { TextValidator, ValidatorForm } from 'react-material-ui-form-validator';

const OnboardingPhoneField = ({ onChange, text, type, value, setUserObject }) => (
    <div className={styles.base}>
        <ValidatorForm onSubmit={() => {}} debounceTime={1000}>
            <TextValidator
                onChange={(e) => onChange({phone: e.target.value, error: false})}
                hintText={value}
                floatingLabelText={text}
                type={type}
                fullWidth={true}
                underlineStyle={{ borderColor: 'white' }}
                underlineFocusStyle={{ borderColor: '#FFB087', opacity: '0.8' }}
                floatingLabelStyle={{ color: 'white', fontSize: '3.6rem', top: '-2rem' }}
                floatingLabelFocusStyle={{ color: 'white', fontSize: '2.4rem' }}
                inputStyle={{ color: 'white', fontSize: '6rem', top: '-2rem', fontWeight: '300', width: '90%' }}
                errorStyle={{ color: 'yellow', position: 'absolute', bottom: '-15px'}}
                value={value}
                name="phone"
                validators={['required']}
                errorMessages={['this field is required']}
                onKeyPress={e => {
                    if ((e.key === 'Enter') && (value !== '')) {
                        setUserObject()
                    }
                }}
            />
        </ValidatorForm>
        <div className={value === '' ? styles.dark : styles.circle} onClick={value === '' ? null : setUserObject}></div>
    </div>
);

export default OnboardingPhoneField;
