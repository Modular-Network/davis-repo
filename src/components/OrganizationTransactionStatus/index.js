/*
 Transaction Component placed in main dash viewhas:
   Tx description, 
   who initiated it, 
   how many signatures have signed it, 
   how many signatures are needed, and 
   status of pending not signed, 
   pending signed, complete, 
   declined, view more button. 
   If pending and not signed, a button to sign or decline, if declined, a button to affirm

   View more component has: TODO
     Tx hash, from address hash, to address hash
*/

import React from 'react'
import styles from './styles.css'

const OrganizationTransactionStatus = ({ data }) => {

  //adding two styles with styles.something syntax?
  return (
    <div className={styles.card}>
      <div className={styles.init}>
        <p>Initiated By: Vera</p>
      </div>
      <div className={styles.transaction}>
        <p className={styles.pending}>(pending)</p>        
        <p className={styles.amount}>0.4 ETH</p>
        <p className={styles.reason}>Well, I really feel we need some cats in the office. Sometimes coding is hard and I want something soft to hold.</p>
      </div>
      <div className='styles.icons'>
        <div>
          <i className='material-icons'>create</i>
          <p>3</p>
        </div>
        <div>
          <i className='material-icons'>not_interested</i>
          <p>1</p>
        </div>

        <div>
          <i className='material-icons'>warning</i>
          <p>2</p>
        </div>

      </div>
      <div className={styles.level}>

        <div>
          <input type="submit" value="Sign" />
        </div>

        <div>
          <input type="submit" value="Decline" />
        </div>

        <div className={styles.circle}>
          <p>5</p>
        </div>

      </div>
    </div>
  )
}

export default OrganizationTransactionStatus;