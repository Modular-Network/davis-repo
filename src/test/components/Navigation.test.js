import React from 'react';
import { shallow } from 'enzyme';
import { Navigation } from 'components/Navigation';

describe('Navigation', () => {

    describe('when user is logged in', () => {

        let navi = shallow(<Navigation auth={true} />);
    
        //console.log(navi.debug());
    
        it('renders properly', () => {
            expect(navi).toMatchSnapshot();
        });
    
        it('contains a connected Appbar component', () => {
            expect(navi.find('Appbar').children('Logged'));
        });
    });

    describe('when user is not logged in', () => {
        
        let navi = shallow(<Navigation auth={false} />);
    
        //console.log(navi.debug());
    
        it('renders properly', () => {
            expect(navi).toMatchSnapshot();
        });
    
        it('contains a connected Appbar component', () => {
            expect(navi.find('Appbar').children('Login'));
        });
    });

});