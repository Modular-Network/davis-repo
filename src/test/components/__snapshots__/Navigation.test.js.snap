// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation when user is logged in renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    auth={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AppBar
        iconElementRight={<Connect(Logged) />}
        showMenuIconButton={true}
        title=""
        zDepth={1}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "iconElementRight": <Connect(Logged) />,
        "showMenuIconButton": true,
        "title": "",
        "zDepth": 1,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AppBar
          iconElementRight={<Connect(Logged) />}
          showMenuIconButton={true}
          title=""
          zDepth={1}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconElementRight": <Connect(Logged) />,
          "showMenuIconButton": true,
          "title": "",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navigation when user is not logged in renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    auth={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AppBar
        iconElementRight={<Connect(Login) />}
        showMenuIconButton={true}
        title=""
        zDepth={1}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "iconElementRight": <Connect(Login) />,
        "showMenuIconButton": true,
        "title": "",
        "zDepth": 1,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AppBar
          iconElementRight={<Connect(Login) />}
          showMenuIconButton={true}
          title=""
          zDepth={1}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconElementRight": <Connect(Login) />,
          "showMenuIconButton": true,
          "title": "",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
